tendermint/src/merkle.rs

```diff
@@ -61,88 +61,31 @@ pub trait MerkleHash {
             },
         }
     }
-}
-
-// A helper to copy GenericArray into the human-friendly Hash type.
-fn copy_to_hash(output: impl AsRef<[u8]>) -> Hash {
-    let mut hash_bytes = [0u8; HASH_SIZE];
-    hash_bytes.copy_from_slice(output.as_ref());
-    hash_bytes
-}
-
-impl<H> MerkleHash for H
-where
-    H: Digest<OutputSize = U32> + FixedOutputReset,
-{
-    fn empty_hash(&mut self) -> Hash {
-        // Get the output of an empty digest state.
-        let digest = self.finalize_reset();
-        copy_to_hash(digest)
-    }
-
-    fn leaf_hash(&mut self, bytes: &[u8]) -> Hash {
-        // Feed the data to the hasher, prepended with 0x00.
-        Digest::update(self, [0x00]);
-        Digest::update(self, bytes);
-
-        // Finalize the digest, reset the hasher state.
-        let digest = self.finalize_reset();
-
-        copy_to_hash(digest)
-    }
 
-    fn inner_hash(&mut self, left: Hash, right: Hash) -> Hash {
-        // Feed the data to the hasher: 0x1, then left and right data.
-        Digest::update(self, [0x01]);
-        Digest::update(self, left);
-        Digest::update(self, right);
-
-        // Finalize the digest, reset the hasher state
-        let digest = self.finalize_reset();
-
-        copy_to_hash(digest)
+    /// Compute a simple Merkle root from vectors of arbitrary byte vectors.
+    /// The leaves of the tree are the bytes of the given byte vectors in
+    /// the given order.
+    fn simple_hash_from_byte_vectors(&mut self, byte_vecs: &[Vec<u8>]) -> Hash {
+        self.hash_byte_vectors(byte_vecs)
     }
 }
 
 /// A wrapper for platform-provided host functions which can't do incremental
 /// hashing. One unfortunate example of such platform is Polkadot.
-pub struct NonIncremental<H>(PhantomData<H>);
-
-impl<H> Default for NonIncremental<H> {
+pub struct DefaultMerkleHash;
```

Comment:
I think this is wrong, we specifically don't want a stateless function to be the basis for default hashing implementation, when in most programming environments we can use the `sha2` crate.

Reply:
I don’t know if I catch you correctly, but as it stands here, it is bounded by `sha2::Sha256`